convert_kinbody_to_urdf.py

```diff
@@ -0,0 +1,194 @@
+                                      features['radius'],
+                                      features['height'])
+                    self.add_cylinder(visual,
+                                      features['radius'],
+                                      features['height'])
+                elif gtype == 'sphere':
+                    self.add_sphere(collision, features['radius'])
+                    self.add_sphere(visual, features['radius'])
+        elif parent.tag.lower() == 'orcdchomp':
+            return
+        else:
+            for child in parent:
+                self.convert_child(child, level + 1)
+
+    def add_sub_element(self, parent, tag, attrib={}, text=''):
+        sub_elem = ET.SubElement(parent, tag)
+        sub_elem.attrib = attrib
+        sub_elem.text = text
+        return sub_elem
+
+    def add_origin(self, parent, val):
+        self.add_sub_element(parent, 'origin', {'xyz': val})
+
+    def add_box(self, parent, val):
+        geo = self.add_sub_element(parent, 'geometry')
+        self.add_sub_element(geo, 'box', {'size': val})
+
+    def add_cylinder(self, parent, radius, length):
+        geo = self.add_sub_element(parent, 'geometry')
+        self.add_sub_element(geo, 'cylinder', {'radius': radius, 'length': length})
+
+    def add_sphere(self, parent, radius):
+        geo = self.add_sub_element(parent, 'geometry')
+        self.add_sub_element(geo, 'sphere', {'radius': radius})
+
+    def add_mesh_render(self, parent, filename):
+        geo = self.add_sub_element(parent, 'geometry')
+        self.add_sub_element(geo, 'mesh', {'filename': filename})
+
+    def add_mesh_data(self, parent, val):
+        ws = val.split(' ')
+        geo = self.add_sub_element(parent, 'geometry')
+        if len(ws) == 1:
+            self.add_sub_element(geo, 'mesh', {'filename': ws[0]})
+        else:
+            self.add_sub_element(geo, 'mesh',
+                                 {'filename': ws[0],
+                                  'scale': '%.2f %.2f %.2f' % (ws[1], ws[1], ws[1])})
+
+    def add_color(self, parent, features):
+        if 'diffusecolor' in features:
+            val = features['diffusecolor']
+        elif 'ambientcolor' in features:
+            val = features['ambientcolor']
+        else:
+            val = '0.2 0.2 0.8'
+
+        if 'transparency' in features:
+            val += ' ' + features['transparency']
+        else:
+            val += ' 1.0'
+
+        mat = self.add_sub_element(parent, 'material', {'name': 'c'})
+        self.add_sub_element(mat, 'color', {'rgba': val})
+
+    def get_urdf(self):
+        if self.urdf is not None:
+            return self.urdf
+
+        self.urdf = ET.ElementTree()
+        urdf_root = ET.Element('robot')
+        urdf_root.attrib = self.kinbody.getroot().attrib
+        urdf_root.text = ''
+        self.link = self.add_sub_element(urdf_root, 'link',
+                                         {'name': 'base_link'})
```

Comment:
addressed in [commit `bb4b8cd`](https://github.com/personalrobotics/pr-ordata/pull/22/commits/bb4b8cdc5232c5187b7e16fdc9892f60ff90f7ef)